defaults/perftune_results.json

```diff
@@ -0,0 +1,179 @@
+        "get-cpu-mask": "0x0000000f",
+        "get-irq-cpu-mask": "0x0000000f",
+        "dump-options-file": {"cpu_mask": "0x0000000f", "irq_cpu_mask": "0x0000000f"}
+      },
+      "8": {
+        "get-cpu-mask": "0x000000fe",
+        "get-irq-cpu-mask": "0x00000001",
+        "dump-options-file": {"cpu_mask": "0x000000ff", "irq_cpu_mask": "0x00000001"}
+      },
+      "12": {
+        "get-cpu-mask": "0x00000fbe",
+        "get-irq-cpu-mask": "0x00000041",
+        "dump-options-file": {"cpu_mask": "0x00000fff", "irq_cpu_mask": "0x00000041"}
+      },
+      "16": {
+        "get-cpu-mask": "0x0000fefe",
+        "get-irq-cpu-mask": "0x00000101",
+        "dump-options-file": {"cpu_mask": "0x0000ffff", "irq_cpu_mask": "0x00000101"}
+      },
+      "24": {
+        "get-cpu-mask": "0x00ffeffe",
+        "get-irq-cpu-mask": "0x00001001",
+        "dump-options-file": {"cpu_mask": "0x00ffffff", "irq_cpu_mask": "0x00001001"}
+      },
+      "32": {
+        "get-cpu-mask": "0xfffefffe",
+        "get-irq-cpu-mask": "0x00010001",
+        "dump-options-file": {"cpu_mask": "0xffffffff", "irq_cpu_mask": "0x00010001"}
+      },
+      "48": {
+        "get-cpu-mask": "0x0000ffff,0xfcfffffc",
+        "get-irq-cpu-mask": "0x03000003",
+        "dump-options-file": {"cpu_mask": "0x0000ffff,0xffffffff", "irq_cpu_mask": "0x03000003"}
+      },
+      "64": {
+        "get-cpu-mask": "0xfffffffc,0xfffffffc",
+        "get-irq-cpu-mask": "0x00000003,0x00000003",
+        "dump-options-file": {"cpu_mask": "0xffffffff,0xffffffff", "irq_cpu_mask": "0x00000003,0x00000003"}
+      },
+      "96": {
+        "get-cpu-mask": "0xfffffcff,0xfffcffff,0xfcfffffc",
+        "get-irq-cpu-mask": "0x00000300,0x00030000,0x03000003",
+        "dump-options-file": {"cpu_mask": "0xffffffff,0xffffffff,0xffffffff", "irq_cpu_mask": "0x00000300,0x00030000,0x03000003"}
+      },
+      "128": {
+        "get-cpu-mask": "0xfffffffc,0xfffffffc,0xfffffffc,0xfffffffc",
+        "get-irq-cpu-mask": "0x00000003,0x00000003,0x00000003,0x00000003",
+        "dump-options-file": {"cpu_mask": "0xffffffff,0xffffffff,0xffffffff,0xffffffff", "irq_cpu_mask": "0x00000003,0x00000003,0x00000003,0x00000003"}
+      }
+    },
+    "aarch64": {
+      "4": {
+        "get-cpu-mask": "0x0000000f",
+        "get-irq-cpu-mask": "0x0000000f",
+        "dump-options-file": {"cpu_mask": "0x0000000f", "irq_cpu_mask": "0x0000000f"}
+      },
+      "8": {
+        "get-cpu-mask": "0x000000fe",
+        "get-irq-cpu-mask": "0x00000001",
+        "dump-options-file": {"cpu_mask": "0x000000ff", "irq_cpu_mask": "0x00000001"}
+      },
+      "16": {
+        "get-cpu-mask": "0x0000fffe",
+        "get-irq-cpu-mask": "0x00000001",
+        "dump-options-file": {"cpu_mask": "0x0000ffff", "irq_cpu_mask": "0x00000001"}
+      },
+      "32": {
+        "get-cpu-mask": "0xfffffffc",
+        "get-irq-cpu-mask": "0x00000003",
+        "dump-options-file": {"cpu_mask": "0xffffffff", "irq_cpu_mask": "0x00000001"}
```

Comment:
This doesn't look right.
In general, I'd expect `get-irq-cpu-mask` and `irq_cpu_mask` be the same.
Why would it be different?

Reply:
True, it seems off, but it is the result I got (instance type `im4gn.8xlarge`):
```
< t:2023-07-23 23:45:08,012 f:remote_base.py  l:520  c:RemoteLibSSH2CmdRunner p:DEBUG > Running command "/opt/scylladb/scripts/perftune.py --tune net --nic eth0 --get-cpu-mask"...
< t:2023-07-23 23:45:08,452 f:base.py         l:228  c:RemoteLibSSH2CmdRunner p:DEBUG > 0xfffffffc
< t:2023-07-23 23:45:08,953 f:base.py         l:142  c:RemoteLibSSH2CmdRunner p:DEBUG > Command "/opt/scylladb/scripts/perftune.py --tune net --nic eth0 --get-cpu-mask" finished with status 0
< t:2023-07-23 23:45:08,953 f:remote_base.py  l:520  c:RemoteLibSSH2CmdRunner p:DEBUG > Running command "/opt/scylladb/scripts/perftune.py --tune net --nic eth0 --get-irq-cpu-mask"...
< t:2023-07-23 23:45:08,955 f:file_logger.py  l:101  c:sdcm.sct_events.file_logger p:INFO  > 2023-07-23 23:45:08.953: (PerftuneResultEvent Severity.NORMAL) period_type=one-time event_id=aeec0a05-c502-41bf-8097-85568a357a99: message=CPU mask commands output: 0xfffffffc
< t:2023-07-23 23:45:09,358 f:base.py         l:228  c:RemoteLibSSH2CmdRunner p:DEBUG > 0x00000003
< t:2023-07-23 23:45:09,858 f:base.py         l:142  c:RemoteLibSSH2CmdRunner p:DEBUG > Command "/opt/scylladb/scripts/perftune.py --tune net --nic eth0 --get-irq-cpu-mask" finished with status 0
< t:2023-07-23 23:45:09,858 f:remote_base.py  l:520  c:RemoteLibSSH2CmdRunner p:DEBUG > Running command "/opt/scylladb/scripts/perftune.py --tune net --nic eth0 --dump-options-file --mode sq_split"...
< t:2023-07-23 23:45:09,860 f:file_logger.py  l:101  c:sdcm.sct_events.file_logger p:INFO  > 2023-07-23 23:45:09.858: (PerftuneResultEvent Severity.NORMAL) period_type=one-time event_id=27da5a5f-ed92-4076-9927-833ec266af88: message=IRQ CPU mask command output: 0x00000003
< t:2023-07-23 23:45:10,177 f:base.py         l:228  c:RemoteLibSSH2CmdRunner p:DEBUG > cpu_mask: '0xffffffff'
< t:2023-07-23 23:45:10,177 f:base.py         l:228  c:RemoteLibSSH2CmdRunner p:DEBUG > irq_core_auto_detection_ratio: 16
< t:2023-07-23 23:45:10,177 f:base.py         l:228  c:RemoteLibSSH2CmdRunner p:DEBUG > irq_cpu_mask: '0x00000001'
< t:2023-07-23 23:45:10,177 f:base.py         l:228  c:RemoteLibSSH2CmdRunner p:DEBUG > nic:
< t:2023-07-23 23:45:10,177 f:base.py         l:228  c:RemoteLibSSH2CmdRunner p:DEBUG > - eth0
< t:2023-07-23 23:45:10,177 f:base.py         l:228  c:RemoteLibSSH2CmdRunner p:DEBUG > tune:
< t:2023-07-23 23:45:10,177 f:base.py         l:228  c:RemoteLibSSH2CmdRunner p:DEBUG > - net
< t:2023-07-23 23:45:10,177 f:base.py         l:228  c:RemoteLibSSH2CmdRunner p:DEBUG > 
< t:2023-07-23 23:45:10,678 f:base.py         l:142  c:RemoteLibSSH2CmdRunner p:DEBUG > Command "/opt/scylladb/scripts/perftune.py --tune net --nic eth0 --dump-options-file --mode sq_split" finished with status 0
```

Reply:
Yes, open an issue.
Mention exactly in which cases we have differences.
If the issue turns out to be invalid we will close it and reference it here.

Reply:
Following your question here (https://github.com/scylladb/seastar/issues/1783#issuecomment-1673309425), @ShlomiBalalis

Reply:
First of all I haven't specified expected outputs for ARM tests in my document.
And your assumption that machines with the same amount of vCPUs on x86 and on ARM is going to produce same masks is baseless.

I have explained to you what's the meaning of `sq` and `sq_splits`, didn't I?

To remind: `sq` is going to dedicate a **single HT** to IRQs while sq_split is going to dedicate a **single CPU core**. And in case of ARM it's exactly the same since they don't have HT to begin with.

As it was also mentioned both by me to you and in corresponding perftune.py patches `--mode` usage has been deprecated. And in case of ARM just like in case of large machines the value of auto-selected `irq_cpu_mask` IS the correct value.

I hope this makes more sense now.

Reply:
@vladzcloudius this thread started by your comment that says the above difference doesn't make sense.
So, does it make sense now?
Do you want to approve it?

Reply:
No, @roydahan the value doesn't make sense and now we understand **why** they were different.
Since, apparently you were generating the `irq_cpu_mask` while enforcing `--mode sq_split` (why?) but you allowed `perftune.py` to auto-detect the `get-irq-cpu-mask`.

And my post above explains why they are coming out different for you.

Please, let me know if it's clearer now, @roydahan?

If not, maybe we should have a quick call.

Reply:
> First of all I haven't specified expected outputs for ARM tests in my document. 

You didn't specify the CPU type at all.

> And your assumption that machines with the same amount of vCPUs on x86 and on ARM is going to produce same masks is baseless.

Where did I make this assumption? I specifically made a different section in the json file to document the results in ARM machines.

> I have explained to you what's the meaning of `sq` and `sq_splits`, didn't I?
> 
> To remind: `sq` is going to dedicate a **single HT** to IRQs while sq_split is going to dedicate a **single CPU core**. And in case of ARM it's exactly the same since they don't have HT to begin with.
> 
> As it was also mentioned both by me to you and in corresponding perftune.py patches `--mode` usage has been deprecated. And in case of ARM just like in case of large machines the value of auto-selected `irq_cpu_mask` IS the correct value.

Under every `Test --dump-options-file` section of your file, you said to use the `--mode` param when executing the command. When we talked, you reiterated what's said in file, and specified: for machines with 0-4 cores, use mq, 5-8 cores, use sq, and 9-32, us sq_split.

Reply:
@ShlomiBalalis we shouldn't specify the value of the mode, we should read it from the configuration that was set.

Reply:
> Under every `Test --dump-options-file` section of your file, you said to use the `--mode` param when executing the command. When we talked, you reiterated what's said in file, and specified: for machines with 0-4 cores, use mq, 5-8 cores, use sq, and 9-32, us sq_split.

@vladzcloudius , could you please clarify it?

Reply:
Nothing to clarify, I already commented with what need to be done.
If not clear, let's set a meeting to clarify everything and stop with this ping pong.